src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Compiler/UnusedTagsPass.php

```diff
@@ -74,6 +74,8 @@ class UnusedTagsPass implements CompilerPassInterface
         'routing.expression_language_provider',
         'routing.loader',
         'routing.route_loader',
+        'routing.service',
+        'routing.variable',
```

Comment:
I suggest to support only services. Adding variables next to services is confusing, increases complexity and has drawbacks that services don't have.